app/src/main/java/de/westnordost/streetcomplete/quests/barrier_opening/AddGateWidthForm.kt

```diff
@@ -0,0 +1,94 @@
+package de.westnordost.streetcomplete.quests.barrier_opening
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isGone
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.databinding.QuestLengthBinding
+import de.westnordost.streetcomplete.osm.Length
+import de.westnordost.streetcomplete.osm.hasDubiousRoadWidth
+import de.westnordost.streetcomplete.quests.AbstractArMeasureQuestForm
+import de.westnordost.streetcomplete.quests.width.WidthAnswer
+import de.westnordost.streetcomplete.screens.measure.ArSupportChecker
+import de.westnordost.streetcomplete.view.controller.LengthInputViewController
+import org.koin.android.ext.android.inject
+
+class AddGateWidthForm : AbstractArMeasureQuestForm<WidthAnswer>() {
+
+    override val contentLayoutResId = R.layout.quest_length
+    private val binding by contentViewBinding(QuestLengthBinding::bind)
+    private val checkArSupport: ArSupportChecker by inject()
+    private var isARMeasurement: Boolean = false
+    private lateinit var lengthInput: LengthInputViewController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        savedInstanceState?.let { isARMeasurement = it.getBoolean(AR) }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val isBarrier = element.tags["barrier"] in GATEWAYS
```

Comment:
When is `isBarrier` false?

Reply:
Sorry, I don't understand - it should be interpreted as a string, and only applies to those items that fit the filters?

Reply:
But the form is only _opened_ for items that fit the filters. So unless I misunderstand something, `isBarrier` should always be true.

Basically, I don't see the reason why this line exists at all. But actually, I don't see the reason why this whole class exists at all. (See comment above.) Can the `AddBarrierOpening` quest not instead use the `AddWidthForm`?

Reply:
You're right, sorry - should be correct now.